code-challenges.test.js

```diff
@@ -14,46 +14,153 @@
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+describe("allCaps", () => {
+  it("The function takes in an array of words and returns an array with all the words capitalized", () => {
+    const randomNouns1 = [
+      "streetlamp",
+      "potato",
+      "teeth",
+      "conclusion",
+      "nephew",
+    ];
+    // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+    const randomNouns2 = ["temperature", "database", "chopsticks", "mango"];
+    // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+    expect(allCaps(randomNouns1)).toEqual([
+      "Streetlamp",
+      "Potato",
+      "Teeth",
+      "Conclusion",
+      "Nephew",
+    ]);
+    expect(allCaps(randomNouns2)).toEqual([
+      "Temperature",
+      "Database",
+      "Chopsticks",
+      "Mango",
+    ]);
+  });
+});
+
+// ReferenceError: allCaps is not defined
 
 // b) Create the function that makes the test pass.
 
+// ============== <<<<< PSEUDO CODE >>>>> ===========================
+// we need a function that takes one parameter. In this scenario that is an array with string elements.
+// we need to iterate through the array and check on each element. For this we are going to use .map function since it returns new array be default.
+// out of every element (string) we need to access the very firs element (index of 0 of [0]) and call .toUpperCase method on that element/value alone
+// final step is adding the rest of the characters using .slice method
+
+const allCaps = (array) => {
+  return array.map((e) => {
+    return e[0].toUpperCase() + e.slice(1);
+  });
+};
+
+//test passed
+
+// ===================================== CHALLENGE #3 =================================================
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
+describe("firstVowel", () => {
+  it("The function that takes in a string and logs the index of the first vowel", () => {
+    const vowelTester1 = "learn";
+    // Expected output: 1
+    const vowelTester2 = "academy";
+    // Expected output: 0
+    const vowelTester3 = "challenges";
+    // Expected output: 2
+    expect(firstVowel(vowelTester1)).toEqual(1);
+    expect(firstVowel(vowelTester2)).toEqual(0);
+    expect(firstVowel(vowelTester3)).toEqual(2);
+  });
+});
+
+//ReferenceError: firstVowel is not defined
 
 // b) Create the function that makes the test pass.
+
+// ============== <<<<< PSEUDO CODE >>>>> ===========================
```

Comment:
Great work in this section!  Your pseudocode is great and logic is solid!  If you want to stretch yourself, try using a higher-order function for this last problem.